### PR TITLE
Add partial support for non-seekable streams (#262)

### PR DESCRIPTION
**Pull request by Toilal**
This is still a work in progress ... I'll update the docs later, but feel free to review this before going further.

### CODE REVIEW

**Comment by nyamsprod** [comment]
The more I think about it the more I'm thinking the seekable property should also be present on the AbstractCsv class too. So that the check is done once ? What do you think ?

File: src/Writer.php [hunk 1]
```diff
@@ -226,6 +226,9 @@ public function addValidator(callable $validator, string $validator_name): self
      */
     public function setNewline(string $newline): self
     {
+        if ("\n" !== $this->newline && $this->document instanceof Stream && !$this->document->isSeekable()) {
```
Comment (nyamsprod) [inline src/Writer.php]: I don't think you need to add this check here... not being able to seek should not prevent setting the newline. This is to be consistent with for instance providing a non writable CSV document to the Writer class. on its own it is not forbidden but you'll get an error on insertion. It should be the same with a non seekable document.

File: src/Stream.php [hunk 2]
```diff
@@ -44,6 +44,13 @@ class Stream implements SeekableIterator
     protected $stream;
 
     /**
+     * Tell whether the stream support seek operations
+     *
+     * @var bool
+     */
+    protected $seekable;
```
Comment (nyamsprod) [inline src/Stream.php]: I would prefer the property to be called is_seekable since it will hold a boolean value.

**Comment by Toilal** [comment]
Do you think this PR really adds value ? It seems to add complexity without adding much value. 

Maybe adding a note in docs to tell user about [this trick](https://github.com/thephpleague/csv/issues/262#issuecomment-332543631) is enough ? I'm using it in my project and it works like a charm.

**Comment by nyamsprod** [comment]
@Toilal If the trick works as expected then yes adding a note somewhere in the documentation is a good enough solution for me. If you can submit a PR with the added info in the documentation I'll gladly merge it.

**Comment by Toilal** [comment]
I'll do, thanks :)

**Pull request closed without merging**